packages/flutter_tools/test/general.shard/runner/target_devices_test.dart

```diff
@@ -2443,11 +2481,21 @@ class TestTargetDevicesWithExtendedWirelessDeviceDiscovery extends TargetDevices
 class TestTargetDeviceSelection extends TargetDeviceSelection {
   TestTargetDeviceSelection(super.logger);
 
-  String input = '';
+  List<String> input = <String>[];
 
   @override
   Future<String> readUserInput() async {
-    return input;
+    // If only one value is provided for the input, continue
+    // to return that one input value without popping
+    //
+    // If more than one input values are provided, we are simulating
+    // the user selecting more than one option for a device, so we will pop
+    // them out from the front
+    if (input.length > 1) {
+      return input.removeAt(0);
+    }
+
+    return input[0];
```

Comment:
hey @vashworth, i reworked this method so that we provide lists of inputs for selecting a device.

Thoughts? The while loop in the `TargetDeviceSelection.userSelectDevice()` method makes it a bit difficult to mock multiple inputs over a test

Reply:
That makes sense to me!